src/components/accession2/edit/CalculatorModal.tsx

```diff
@@ -33,17 +33,40 @@ export default function CalculatorModal(props: CalculatorModalProps): JSX.Elemen
   const classes = useStyles();
   const snackbar = useSnackbar();
   const [subsetError, setSubsetError] = useState('');
+  const [subsetWeightError, setSubsetWeightError] = useState('');
+  const [totalWeightError, setTotalWeightError] = useState('');
+  const [subsetCountError, setSubsetCountError] = useState('');
   const preferredUnits = usePreferredWeightUnits();
 
   const validateFields = () => {
-    if (record.subsetWeight?.units === record.remainingQuantity?.units) {
+    let hasErrors = false;
+
+    setSubsetError('');
+    setSubsetWeightError('');
+    if (!record.subsetWeight?.quantity || !record.subsetWeight?.units) {
+      setSubsetError(strings.REQUIRED_VALUE_AND_UNITS_FIELD);
+      hasErrors = true;
+    } else if (record.subsetWeight?.units === record.remainingQuantity?.units) {
       if (+(record.subsetWeight?.quantity || 0) > +(record.remainingQuantity?.quantity || 0)) {
-        setSubsetError(strings.SUBSET_ERROR);
-        return false;
+        setSubsetWeightError(strings.SUBSET_WEIGHT_ERROR);
+        hasErrors = true;
       }
     }
-    setSubsetError('');
-    return true;
+
+    setSubsetCountError('');
+    if (!record.subsetCount) {
+      setSubsetCountError(strings.REQUIRED_FIELD);
+      hasErrors = true;
+    }
+
+    setTotalWeightError('');
+    const totalWeightUnits = record.remainingQuantity?.units ?? 'Seeds';
```

Comment:
Why are we settings 'seeds' here and not just undefined if no unit is selected? Then we could just check if totalWeightUnits.  I don't think it can we undefined anyways

Reply:
yes, it defaults to Seeds.. i possibly don't understand this code very well

Reply:
changing this check to `(!totalWeightUnits || totalWeightUnits === 'Seeds')` even though it can't be undefined theoretically but according to the typing it can be